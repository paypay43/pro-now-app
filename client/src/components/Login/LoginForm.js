import React, { useState } from 'react';
import TokenService from '../../services/token-service';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import AuthApiService from '../../services/auth-api-service';
import UserContext from '../../contexts/UserContext';
import './Login.css';

export default function LoginForm(props) {
  const userContext = React.useContext(UserContext);
  ;
  const [error, setError] = useState(null);

  const handleSubmitJwtAuth = ev => {
    ;
    ev.preventDefault();
    setError(null);
    const { username, password } = ev.target;

    AuthApiService.postLogin({
      username: username.value,
      password: password.value
    })
      .then(res => {
        ;
        username.value = '';
        password.value = '';
        TokenService.saveAuthToken(res.authToken);
        userContext.setLoggedIn(true);
        props.onLoginSuccess();
        ;
      })
      .catch(res => {
        ;
        setError({ error: res.error });
      });
  };

  const displayError = () => {
    return Object.keys(error).map(e => {
      return (
        <span key={e}>
          {e}: {error[e]}
        </span>
      );
    });
  };

  return (
    <form className="login-form app-form" onSubmit={handleSubmitJwtAuth}>
      <FontAwesomeIcon
        className="black big-icon form-icon"
        icon="fist-raised"
      />
      <h1>Please sign in</h1>
      <div role="alert">{error && <p className="red">{displayError()}</p>}</div>

      <input
        type="username"
        name="username"
        id="inputUsername"
        class="form-control"
        placeholder="Username"
        required=""
        autofocus=""
      />

      <input
        type="password"
        name="password"
        id="inputPassword"
        class="form-control"
        placeholder="Password"
        required=""
      />
      <button className="btn" type="submit">
        Login
      </button>
    </form>
  );
}
