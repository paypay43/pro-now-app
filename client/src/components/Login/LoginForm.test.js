import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';

describe(`LoginForm component`, () => {
  it('renders loginform', () => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoginForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
