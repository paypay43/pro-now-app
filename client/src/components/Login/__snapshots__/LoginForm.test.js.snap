// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders loginform 1`] = `
<form
  className="login-form app-form"
  onSubmit={[Function]}
>
  <FontAwesomeIcon
    border={false}
    className="black big-icon form-icon"
    fixedWidth={false}
    flip={null}
    icon="fist-raised"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <h1>
    Please sign in
  </h1>
  <div
    role="alert"
  />
  <input
    autofocus=""
    class="form-control"
    id="inputUsername"
    name="username"
    placeholder="Username"
    required=""
    type="username"
  />
  <input
    class="form-control"
    id="inputPassword"
    name="password"
    placeholder="Password"
    required=""
    type="password"
  />
  <button
    className="btn"
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`NotefulForm component renders loginform 1`] = `
<form
  className="login-form app-form"
  onSubmit={[Function]}
>
  <FontAwesomeIcon
    border={false}
    className="black big-icon form-icon"
    fixedWidth={false}
    flip={null}
    icon="fist-raised"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <h1>
    Please sign in
  </h1>
  <div
    role="alert"
  />
  <input
    autofocus=""
    class="form-control"
    id="inputUsername"
    name="username"
    placeholder="Username"
    required=""
    type="username"
  />
  <input
    class="form-control"
    id="inputPassword"
    name="password"
    placeholder="Password"
    required=""
    type="password"
  />
  <button
    className="btn"
    type="submit"
  >
    Login
  </button>
</form>
`;
